gaffer-as-a-service/gaas-rest/pom.xml

```diff
@@ -73,6 +73,20 @@
             <artifactId>spring-security-oauth2-jose</artifactId>
             <version>5.6.0</version>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter</artifactId>
+            <exclusions>
+                <exclusion>
+                    <artifactId>log4j-to-slf4j</artifactId>
+                    <groupId>org.apache.logging.log4j</groupId>
+                </exclusion>
+                <exclusion>
+                    <artifactId>log4j-api</artifactId>
+                    <groupId>org.apache.logging.log4j</groupId>
+                </exclusion>
```

Comment:
There also needs to be an exclusion for `ch.qos.logback:logback-classic`. As is, the problem with the SLF4J logger conflict is still occurring. [Can be seen here in the CI for this PR.](https://github.com/gchq/gaffer-experimental/runs/5836829692?check_suite_focus=true#step:6:278)

This will be quite complex to resolve as Logback is in fact being used in [DeploymentHandlerTest.java](https://github.com/gchq/gaffer-experimental/blob/master/gaffer-as-a-service/gaas-rest/src/test/java/uk/gov/gchq/gaffer/gaas/handlers/DeploymentHandlerTest.java). Either the use of Logback in that file needs to be removed or the project changed to Logback instead of Reload4j. It's unclear why Logback was used in that file when it was not being used anywhere else.

Reply:
I feel like the easiest way may be to just use slf4j instead of logback directly?

Reply:
I agree the best solution is to not use Logback, although there's no way to use SLF4J as it is just a translation layer with no way to add appenders. Log4j (implemented by Reload4j) would need to be used directly. The changes required to the code using Logback are quite simple, although a custom appender needs to be defined as Log4j doesn't have one built in. The inability to do this sort of testing through SLF4J may be one of the reasons why Gaffer doesn't have this kind of test.

Reply:
Hmm that is annoying

Reply:
@GCHQDeveloper314 , @t92549 
Excluded **ch.qos.logback:logback-classic** and  used **Reload4j** instead of Logback by [DeploymentHandlerTest.java](https://github.com/gchq/gaffer-experimental/blob/master/gaffer-as-a-service/gaas-rest/src/test/java/uk/gov/gchq/gaffer/gaas/handlers/DeploymentHandlerTest.java).

Reply:
That fix is pretty much what I had drafted out locally. This is resolved